frontend/src/pages/notebookController/screens/admin/NotebookAdminControl.tsx

```diff
@@ -16,16 +16,43 @@ import useAdminUsers from './useAdminUsers';
 import ExternalLink from '../../../../components/ExternalLink';
 import useTableColumnSort from '../../../../utilities/useTableColumnSort';
 import { User } from './types';
+import StopServerModal from '../server/StopServerModal';
+import { NotebookAdminContext } from './NotebookAdminContext';
+import { Notebook } from '../../../../types';
 
 const INITIAL_PAGE_LIMIT = 10;
 const NotebookAdminControl: React.FC = () => {
   const [unsortedUsers, loaded, loadError] = useAdminUsers();
   const [pageIndex, setPageIndex] = React.useState(0);
   const [perPage, setPerPage] = React.useState(INITIAL_PAGE_LIMIT);
   const { transformData, getColumnSort } = useTableColumnSort<User>(columns, 0);
+  const { serverStatuses, setServerStatuses } = React.useContext(NotebookAdminContext);
 
   const users = transformData(unsortedUsers);
 
+  const onNotebooksStop = React.useCallback(
+    (didStop: boolean) => {
+      if (didStop) {
+        serverStatuses.forEach((serverStatus) => {
+          serverStatus.forceRefresh();
+        });
+      }
+      setServerStatuses([]);
+    },
+    [serverStatuses, setServerStatuses],
+  );
+
+  const notebooksToStop = React.useMemo(
+    () =>
+      serverStatuses.reduce((prev, curr) => {
+        if (curr.notebook) {
+          prev.push(curr.notebook);
+        }
+        return prev;
+      }, [] as Notebook[]),
+    [serverStatuses],
```

Comment:
Just do `.filter().map()`... `.reduce()` is going to complicate the implementation. 

Also, kudos on the memo, but it's already set state filtering driven -- not sure we need to do that. I'm not against it though. Computing on each render isn't that crazy for a component.